src/main/java/com/android/volley/toolbox/DiskBasedCacheUtility.java

```diff
@@ -37,6 +43,95 @@ static String getFilenameForKey(String key) {
         return localFilename;
     }
 
+    /** Returns a file object for the given cache key. */
+    public static File getFileForKey(String key, FileSupplier rootDirectorySupplier) {
+        return new File(rootDirectorySupplier.get(), getFilenameForKey(key));
+    }
+
+    static boolean wouldExceedCacheSize(long newTotalSize, int maxCacheSize) {
+        return (newTotalSize >= maxCacheSize);
+    }
+
+    static boolean isDataTooLarge(int dataLength, int maxCacheSize) {
+        return dataLength >= maxCacheSize * HYSTERESIS_FACTOR;
+    }
+
+    /**
+     * Prunes the cache if needed. This method modifies the entries map by removing the pruned
+     * entries.
+     *
+     * @param totalSize The total size of the cache.
+     * @param maxCacheSizeInBytes Maximum size of the cache.
+     * @param entries Map of the entries in the cache.
+     * @param rootDirectorySupplier The supplier for the root directory to use for the cache.
+     * @return The updated totalSize.
+     */
+    static long pruneIfNeeded(
+            long totalSize,
+            int maxCacheSizeInBytes,
+            Map<String, CacheHeader> entries,
+            FileSupplier rootDirectorySupplier) {
+        if (!wouldExceedCacheSize(totalSize, maxCacheSizeInBytes)) {
+            return totalSize;
+        }
+        if (VolleyLog.DEBUG) {
+            VolleyLog.v("Pruning old cache entries.");
+        }
+
+        long before = totalSize;
+        int prunedFiles = 0;
+        long startTime = SystemClock.elapsedRealtime();
+
+        Iterator<Map.Entry<String, CacheHeader>> iterator = entries.entrySet().iterator();
+        while (iterator.hasNext()) {
+            Map.Entry<String, CacheHeader> entry = iterator.next();
+            CacheHeader e = entry.getValue();
+            boolean deleted = getFileForKey(e.key, rootDirectorySupplier).delete();
+            if (deleted) {
+                totalSize -= e.size;
+            } else {
+                VolleyLog.d(
+                        "Could not delete cache entry for key=%s, filename=%s",
+                        e.key, getFilenameForKey(e.key));
+            }
+            iterator.remove();
+            prunedFiles++;
+
+            if (!isDataTooLarge((int) totalSize, maxCacheSizeInBytes)) {
```

Comment:
Casting a `long` to `int` is a lossy conversion: if the value of the long needs more than 32 bits, some of them will be discarded to fit in the `int`.

It'd be safer for `isDataTooLarge` to take a `long` instead (so you'd be casting an `int` to `long`, which is fine)

Reply:
oh yeah, that makes sense. thanks!